### ISSUE TITLE
Prerender with named html files

### ISSUE THREAD

**Issue by slyduda** [unjs/nitro#607]
### Describe the feature

The prerenderer currently prefers prerendering in subfolders. While I can take the advice of writing named html files in the prerender route config, I would prefer to make this the togglable behavior with a bool in the nitro config (similar to how there was one in Nuxt v2).

After doing some searching this was the default behavior before [#166](https://github.com/unjs/nitro/issues/166). It would be nice to have a bool that toggles this in nitro config if it does not already exist.

### Additional information

- [x] Would you be willing to help implement this feature?

**Comment by emretepedev** [comment]
Same issue here.

I don't want to create a folder for each page. This behavior must be changeable via config.

**Comment by davidsandoz** [comment]
I would need this as well.

As already mentioned, this used to be the default before it got changed in cdcd35adf3e4f1d3902605f97559c0dcf6d0febf via #166.

In nuxt/nuxt#13886 (where the opposite was requested, before #166 got implemented), a workaround was mentioned in [this comment](https://github.com/nuxt/nuxt/issues/13886#issuecomment-1397320165) by @pi0:

> This is now the default behavior of Nuxt 3 in latest RC to use dist/[path]/index.html (unless path has an explicit .html extension)

And indeed, if we name a page component `foo.html.vue`, then we get the file `foo.html` in `.output/public/`.

However, I noticed a problem with this workaround. The generated file has references to `/foo.html/_payload.js`:
```html
<link rel="preload" as="fetch" crossorigin="anonymous" href="/foo.html/_payload.js">
```
```html
<script type="module">import p from "/foo.html/_payload.js";window.__NUXT__={...p,...({state:{},_errors:{},serverRendered:true,prerenderedAt:1681384634837,config:{public:{},app:{baseURL:"\u002F",buildAssetsDir:"\u002F_nuxt\u002F",cdnURL:""}}})}</script>
```

As `foo.html` is not a directory, it fails at retrieving the `_payload.js` file.

Is there a way to configure this properly? Or is this something that has to be fixed in nitro or Nuxt?

**Comment by davidsandoz** [comment]
Additionally, if I have a page component with a parameter, e.g. `bar-[id].html.vue`, then the link crawler doesn't generate the page even though there is the following nuxt-link in `index.vue`:
```html
<nuxt-link to="bar-3.html">Bar 3</nuxt-link>
```
And there are no errors during generation.

While if I put `bar-3` instead of `bar-3.html`, I get the following error:
```
/bar-3 (7ms) (Error: [404] Page not found: /bar-3)  
```

Here's a repo with minimal changes to reproduce the issue: https://github.com/davidsandoz/nuxt-starter-static-payload-reproduction/tree/v3-codesandbox-static-payload-reproduction

**Comment by Hebilicious** [comment]
@pi0 Can this be closed thanks to #1676 ?
